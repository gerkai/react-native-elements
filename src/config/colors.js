import { StyleSheet } from 'react-native';

export default {
  primary: '#2089dc',
  secondary: '#8F0CE8',
  grey0: '#393e42',
  grey1: '#43484d',
  grey2: '#5e6977',
  grey3: '#86939e',
  grey4: '#bdc6cf',
  grey5: '#e1e8ee',
  greyOutline: '#bbb',
  searchBg: '#303337',
  error: '#ff190c',
  disabled: 'hsl(208, 8%, 90%)',
  // Darker color if hairlineWidth is not thin enough
  divider: StyleSheet.hairlineWidth < 1 ? '#bcbbc1' : 'rgba(0, 0, 0, 0.12)',
};
