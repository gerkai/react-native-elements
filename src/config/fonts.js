export default {
  ios: {
  },
  android: {
    regular: 'sans-serif',
    light: 'sans-serif-light',
    condensed: 'sans-serif-condensed',
    condensed_light: 'sans-serif-condensed-light',
    black: 'sans-serif-black',
    thin: 'sans-serif-thin',
    medium: 'sans-serif-medium',
    bold: 'sans-serif-bold',
  }
}
