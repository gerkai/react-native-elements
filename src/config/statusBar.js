import { Dimensions, Platform, StatusBar } from 'react-native';

// Borrowed from https://github.com/ptelad/react-native-iphone-x-helper

function isIphoneX() {
  const dimen = Dimensions.get('window');
  return (
    Platform.OS === 'ios' &&
    !Platform.isPad &&
    !Platform.isTVOS &&
    (dimen.height === 812 || dimen.width === 812)
  );
}

function ifIphoneX(iphoneXStyle, regularStyle) {
  if (isIphoneX()) {
    return iphoneXStyle;
  }
  return regularStyle;
}

export function getStatusBarHeight(skipAndroid = false) {
  if (Platform.OS === 'ios') {
    return ifIphoneX(44, 20);
  }

  if (skipAndroid) {
    return 0;
  }

  return StatusBar.currentHeight;
}
