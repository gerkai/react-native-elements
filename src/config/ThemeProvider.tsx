import React from 'react';
import deepmerge from 'deepmerge';
import colors from './colors';
import darkColors from './colorsDark';
import { FullTheme, Theme } from './theme';

type RecursivePartial<T> = { [P in keyof T]?: RecursivePartial<T[P]> };

export interface ThemeProps<T> {
  theme: Theme<T>;
  updateTheme: (updates: RecursivePartial<FullTheme>) => void;
  replaceTheme: (updates: RecursivePartial<FullTheme>) => void;
}

export const ThemeContext: React.Context<ThemeProps<{}>> = React.createContext({
  theme: {
    colors,
  },
} as ThemeProps<{}>);

export type ThemeProviderProps = {
  useDark?: boolean;
};

type ThemeProviderState = {
  theme: Theme;
  useDark: boolean;
};

export default class ThemeProvider extends React.Component<
  ThemeProviderProps,
  ThemeProviderState
> {
  static defaultProps = {
    theme: {},
    useDark: false,
  };
  defaultTheme: Partial<FullTheme>;

  constructor(props: { theme: Theme; useDark?: boolean }) {
    super(props);
    const defaultColors = props.useDark ? darkColors : colors;
    this.defaultTheme = deepmerge(
      {
        colors: defaultColors,
      },
      props.theme
    );
    this.state = {
      theme: this.defaultTheme,
      useDark: Boolean(props.useDark),
    };
  }

  static getDerivedStateFromProps(
    props: {
      theme: Theme;
      useDark?: boolean;
    },
    state: ThemeProviderState
  ) {
    const { useDark } = props;
    if (useDark !== state.useDark) {
      const defaultColors = useDark ? darkColors : colors;
      return {
        theme: deepmerge(
          state.theme,
          deepmerge(
            {
              colors: defaultColors,
            },
            props.theme
          )
        ),
        useDark,
      };
    }
    return null;
  }

  updateTheme = (updates: RecursivePartial<FullTheme>) => {
    this.setState(({ theme }) => ({
      theme: deepmerge(theme, updates),
    }));
  };

  replaceTheme = (theme: RecursivePartial<FullTheme>) => {
    this.setState(() => ({
      theme: deepmerge(this.defaultTheme, theme),
    }));
  };

  getTheme = () => this.state.theme;

  render() {
    return (
      <ThemeContext.Provider
        value={{
          theme: this.state.theme,
          updateTheme: this.updateTheme,
          replaceTheme: this.replaceTheme,
        }}
      >
        {this.props.children}
      </ThemeContext.Provider>
    );
  }
}

export const ThemeConsumer = ThemeContext.Consumer;
