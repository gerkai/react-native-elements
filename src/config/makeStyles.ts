import { useContext } from 'react';
import { StyleSheet } from 'react-native';
import { FullTheme } from './theme';
import { ThemeContext } from './ThemeProvider';

export const useTheme = () => {
  return useContext(ThemeContext);
};

export const makeStyles = <
  T extends StyleSheet.NamedStyles<T> | StyleSheet.NamedStyles<any>,
  V
>(
  styles: T | ((theme: Partial<FullTheme>, props: V) => T)
) => (props: V = {} as any): T => {
  const { theme } = useTheme();
  const css = typeof styles === 'function' ? styles(theme, props) : styles;

  return StyleSheet.create(css);
};
