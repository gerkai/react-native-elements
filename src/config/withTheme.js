import React from 'react';
import merge from 'lodash.merge';
import { ThemeConsumer } from './ThemeProvider';
import DefaultTheme from './theme';

const isClassComponent = Component =>
  Boolean(Component.prototype && Component.prototype.isReactComponent);

const withTheme = (WrappedComponent, themeKey) => {
  class ThemedComponent extends React.Component {
    render() {
      /* eslint-disable react/prop-types */
      const { forwardedRef, ...rest } = this.props;

      return (
        <ThemeConsumer>
          {context => {
            // If user isn't using ThemeProvider
            if (!context) {
              let props = { ...rest, theme: DefaultTheme };

              return isClassComponent(WrappedComponent) ? (
                <WrappedComponent ref={forwardedRef} {...props} />
              ) : (
                <WrappedComponent {...props} />
              );
            }

            const { theme, updateTheme } = context;
            const props = {
              theme,
              updateTheme,
              ...merge({}, themeKey && theme[themeKey], rest),
            };

            if (isClassComponent(WrappedComponent)) {
              return <WrappedComponent ref={forwardedRef} {...props} />;
            }
            return <WrappedComponent {...props} />;
          }}
        </ThemeConsumer>
      );
    }
  }

  const name = themeKey
    ? `Themed.${themeKey}`
    : `Themed.${WrappedComponent.displayName ||
        WrappedComponent.name ||
        'Component'}`;

  const forwardRef = (props, ref) => {
    return <ThemedComponent {...props} forwardedRef={ref} />;
  };

  ThemedComponent.displayName = name;
  forwardRef.displayName = name;

  // Forward refs from children
  if (isClassComponent(WrappedComponent)) {
    return React.forwardRef(forwardRef);
  }

  return ThemedComponent;
};

export default withTheme;
