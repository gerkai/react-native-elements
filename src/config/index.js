import { Text } from 'react-native';
import BackgroundImage from './BackgroundImage';
import colors from './colors';
import { getStatusBarHeight } from './statusBar';
import ViewPropTypes from './ViewPropTypes';
import fonts from './fonts';
import ThemeProvider, { ThemeConsumer } from './ThemeProvider';
import merge from 'lodash.merge';
import withTheme from './withTheme';

const TextPropTypes = Text.propTypes;

export {
  BackgroundImage,
  colors,
  getStatusBarHeight,
  ViewPropTypes,
  TextPropTypes,
  fonts,
  ThemeProvider,
  ThemeConsumer,
  merge,
  withTheme,
};
