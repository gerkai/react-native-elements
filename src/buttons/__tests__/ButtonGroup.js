import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { create } from 'react-test-renderer';

import theme from '../../config/theme';
import { ThemeProvider } from '../../config';

import ThemedButtonGroup, { ButtonGroup } from '../ButtonGroup';

const buttons = ['Button 1', 'Button 2', 'Button 3'];

describe('ButtonGroup Component', () => {
  it('should render without issues', () => {
    const component = shallow(<ButtonGroup theme={theme} buttons={buttons} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should have onPress event', () => {
    const onPress = jest.fn();
    const component = shallow(
      <ButtonGroup
        theme={theme}
        buttons={buttons}
        onPress={onPress}
        containerStyle={{ backgroundColor: 'yellow' }}
        buttonStyle={{ backgroundColor: 'blue' }}
      />
    );

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render selectedIndex', () => {
    const component = shallow(
      <ButtonGroup
        theme={theme}
        buttons={buttons}
        selectedIndex={1}
        selectedButtonStyle={{ backgroundColor: 'red' }}
        selectedTextStyle={{ fontSize: 12 }}
      />
    );

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render with button.element', () => {
    const component = shallow(
      <ButtonGroup
        theme={theme}
        buttons={[{ element: 'Text' }, { element: 'View' }]}
        innerBorderStyle={{ width: 300, color: 'red' }}
      />
    );

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render lastButtonStyle', () => {
    const component = shallow(
      <ButtonGroup
        theme={theme}
        buttons={buttons}
        lastBorderStyle={{ backgroundColor: 'red' }}
      />
    );

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render without inner borders', () => {
    const component = shallow(
      <ButtonGroup
        theme={theme}
        buttons={buttons}
        innerBorderStyle={{ width: 0 }}
      />
    );

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should select an item', () => {
    const onPress = jest.fn();

    const wrapper = shallow(
      <ButtonGroup
        theme={theme}
        buttons={buttons}
        innerBorderStyle={{ width: 0 }}
        onPress={onPress}
      />
    );

    wrapper
      .find({ testID: 'buttonGroupItem' })
      .at(2)
      .simulate('press');
    expect(onPress).toBeCalledWith(2);
  });

  describe('Select Multiple', () => {
    it('should select an additional item', () => {
      const onPress = jest.fn();

      const wrapper = shallow(
        <ButtonGroup
          theme={theme}
          buttons={buttons}
          innerBorderStyle={{ width: 0 }}
          selectMultiple
          selectedIndexes={[0]}
          onPress={onPress}
        />
      );

      wrapper
        .find({ testID: 'buttonGroupItem' })
        .at(2)
        .simulate('press');
      expect(onPress).toBeCalledWith([0, 2]);
    });

    it('should deselect a selected item', () => {
      const onPress = jest.fn();

      const wrapper = shallow(
        <ButtonGroup
          theme={theme}
          buttons={buttons}
          innerBorderStyle={{ width: 0 }}
          selectMultiple
          selectedIndexes={[0, 2]}
          onPress={onPress}
        />
      );

      wrapper
        .find({ testID: 'buttonGroupItem' })
        .at(2)
        .simulate('press');
      expect(onPress).toBeCalledWith([0]);
    });
  });

  it('should apply values from theme', () => {
    const theme = {
      ButtonGroup: {
        selectedTextStyle: {
          color: 'red',
        },
      },
    };

    const component = create(
      <ThemeProvider theme={theme}>
        <ThemedButtonGroup buttons={buttons} selectedIndex={0} />
      </ThemeProvider>
    );

    expect(
      component.root.findAllByProps({ testID: 'buttonGroupItemText' })[0].props
        .style
    ).toMatchObject({
      color: 'red',
    });
    expect(component.toJSON()).toMatchSnapshot();
  });
});
