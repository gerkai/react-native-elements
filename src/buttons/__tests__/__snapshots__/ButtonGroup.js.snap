// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup Component should apply values from theme 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderColor": "#e3e3e3",
      "borderRadius": 3,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 40,
      "marginBottom": 5,
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 5,
      "overflow": "hidden",
    }
  }
  updateTheme={[Function]}
>
  <View
    style={
      Object {
        "borderBottomLeftRadius": 3,
        "borderRightColor": "#bdc6cf",
        "borderRightWidth": 0,
        "borderTopLeftRadius": 3,
        "flex": 1,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2089dc",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "red",
              "fontSize": 16.25,
              "fontWeight": "500",
            }
          }
          testID="buttonGroupItemText"
          theme={
            Object {
              "ButtonGroup": Object {
                "selectedTextStyle": Object {
                  "color": "red",
                },
              },
              "colors": Object {
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
              },
            }
          }
          updateTheme={[Function]}
        >
          Button 1
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "borderLeftColor": "#bdc6cf",
        "borderLeftWidth": 1,
        "borderRightColor": "#bdc6cf",
        "borderRightWidth": 1,
        "flex": 1,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#5e6977",
              "fontSize": 16.25,
              "fontWeight": "500",
            }
          }
          testID="buttonGroupItemText"
          theme={
            Object {
              "ButtonGroup": Object {
                "selectedTextStyle": Object {
                  "color": "red",
                },
              },
              "colors": Object {
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
              },
            }
          }
          updateTheme={[Function]}
        >
          Button 2
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "borderBottomRightRadius": 3,
        "borderTopRightRadius": 3,
        "flex": 1,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#5e6977",
              "fontSize": 16.25,
              "fontWeight": "500",
            }
          }
          testID="buttonGroupItemText"
          theme={
            Object {
              "ButtonGroup": Object {
                "selectedTextStyle": Object {
                  "color": "red",
                },
              },
              "colors": Object {
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
              },
            }
          }
          updateTheme={[Function]}
        >
          Button 3
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`ButtonGroup Component should have onPress event 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "yellow",
      "borderColor": "#e3e3e3",
      "borderRadius": 3,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 40,
      "marginBottom": 5,
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 5,
      "overflow": "hidden",
    }
  }
>
  <Component
    key="0"
    style={
      Object {
        "borderBottomLeftRadius": 3,
        "borderRightColor": "#bdc6cf",
        "borderRightWidth": 0,
        "borderTopLeftRadius": 3,
        "flex": 1,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      disabled={false}
      onPress={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
      underlayColor="#2089dc"
    >
      <Component
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "blue",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Themed.Text
          style={
            Object {
              "color": "#5e6977",
              "fontSize": 16.25,
              "fontWeight": "500",
            }
          }
          testID="buttonGroupItemText"
        >
          Button 1
        </Themed.Text>
      </Component>
    </TouchableOpacity>
  </Component>
  <Component
    key="1"
    style={
      Object {
        "borderLeftColor": "#bdc6cf",
        "borderLeftWidth": 1,
        "borderRightColor": "#bdc6cf",
        "borderRightWidth": 1,
        "flex": 1,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      disabled={false}
      onPress={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
      underlayColor="#2089dc"
    >
      <Component
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "blue",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Themed.Text
          style={
            Object {
              "color": "#5e6977",
              "fontSize": 16.25,
              "fontWeight": "500",
            }
          }
          testID="buttonGroupItemText"
        >
          Button 2
        </Themed.Text>
      </Component>
    </TouchableOpacity>
  </Component>
  <Component
    key="2"
    style={
      Object {
        "borderBottomRightRadius": 3,
        "borderTopRightRadius": 3,
        "flex": 1,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      disabled={false}
      onPress={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
      underlayColor="#2089dc"
    >
      <Component
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "blue",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Themed.Text
          style={
            Object {
              "color": "#5e6977",
              "fontSize": 16.25,
              "fontWeight": "500",
            }
          }
          testID="buttonGroupItemText"
        >
          Button 3
        </Themed.Text>
      </Component>
    </TouchableOpacity>
  </Component>
</Component>
`;

exports[`ButtonGroup Component should render lastButtonStyle 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#fff",
      "borderColor": "#e3e3e3",
      "borderRadius": 3,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 40,
      "marginBottom": 5,
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 5,
      "overflow": "hidden",
    }
  }
>
  <Component
    key="0"
    style={
      Object {
        "borderBottomLeftRadius": 3,
        "borderRightColor": "#bdc6cf",
        "borderRightWidth": 0,
        "borderTopLeftRadius": 3,
        "flex": 1,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      disabled={false}
      onPress={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
      underlayColor="#2089dc"
    >
      <Component
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Themed.Text
          style={
            Object {
              "color": "#5e6977",
              "fontSize": 16.25,
              "fontWeight": "500",
            }
          }
          testID="buttonGroupItemText"
        >
          Button 1
        </Themed.Text>
      </Component>
    </TouchableOpacity>
  </Component>
  <Component
    key="1"
    style={
      Object {
        "borderLeftColor": "#bdc6cf",
        "borderLeftWidth": 1,
        "borderRightColor": "#bdc6cf",
        "borderRightWidth": 1,
        "flex": 1,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      disabled={false}
      onPress={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
      underlayColor="#2089dc"
    >
      <Component
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Themed.Text
          style={
            Object {
              "color": "#5e6977",
              "fontSize": 16.25,
              "fontWeight": "500",
            }
          }
          testID="buttonGroupItemText"
        >
          Button 2
        </Themed.Text>
      </Component>
    </TouchableOpacity>
  </Component>
  <Component
    key="2"
    style={
      Object {
        "backgroundColor": "red",
        "borderBottomRightRadius": 3,
        "borderTopRightRadius": 3,
        "flex": 1,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      disabled={false}
      onPress={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
      underlayColor="#2089dc"
    >
      <Component
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Themed.Text
          style={
            Object {
              "color": "#5e6977",
              "fontSize": 16.25,
              "fontWeight": "500",
            }
          }
          testID="buttonGroupItemText"
        >
          Button 3
        </Themed.Text>
      </Component>
    </TouchableOpacity>
  </Component>
</Component>
`;

exports[`ButtonGroup Component should render selectedIndex 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#fff",
      "borderColor": "#e3e3e3",
      "borderRadius": 3,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 40,
      "marginBottom": 5,
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 5,
      "overflow": "hidden",
    }
  }
>
  <Component
    key="0"
    style={
      Object {
        "borderBottomLeftRadius": 3,
        "borderRightColor": "#bdc6cf",
        "borderRightWidth": 0,
        "borderTopLeftRadius": 3,
        "flex": 1,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      disabled={false}
      onPress={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
      underlayColor="#2089dc"
    >
      <Component
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Themed.Text
          style={
            Object {
              "color": "#5e6977",
              "fontSize": 16.25,
              "fontWeight": "500",
            }
          }
          testID="buttonGroupItemText"
        >
          Button 1
        </Themed.Text>
      </Component>
    </TouchableOpacity>
  </Component>
  <Component
    key="1"
    style={
      Object {
        "borderLeftColor": "#bdc6cf",
        "borderLeftWidth": 1,
        "borderRightColor": "#bdc6cf",
        "borderRightWidth": 1,
        "flex": 1,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      disabled={false}
      onPress={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
      underlayColor="#2089dc"
    >
      <Component
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "red",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Themed.Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 12,
              "fontWeight": "500",
            }
          }
          testID="buttonGroupItemText"
        >
          Button 2
        </Themed.Text>
      </Component>
    </TouchableOpacity>
  </Component>
  <Component
    key="2"
    style={
      Object {
        "borderBottomRightRadius": 3,
        "borderTopRightRadius": 3,
        "flex": 1,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      disabled={false}
      onPress={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
      underlayColor="#2089dc"
    >
      <Component
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Themed.Text
          style={
            Object {
              "color": "#5e6977",
              "fontSize": 16.25,
              "fontWeight": "500",
            }
          }
          testID="buttonGroupItemText"
        >
          Button 3
        </Themed.Text>
      </Component>
    </TouchableOpacity>
  </Component>
</Component>
`;

exports[`ButtonGroup Component should render with button.element 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#fff",
      "borderColor": "#e3e3e3",
      "borderRadius": 3,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 40,
      "marginBottom": 5,
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 5,
      "overflow": "hidden",
    }
  }
>
  <Component
    key="0"
    style={
      Object {
        "borderBottomLeftRadius": 3,
        "borderRightColor": "red",
        "borderRightWidth": 0,
        "borderTopLeftRadius": 3,
        "flex": 1,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      disabled={false}
      onPress={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
      underlayColor="#2089dc"
    >
      <Component
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text />
      </Component>
    </TouchableOpacity>
  </Component>
  <Component
    key="1"
    style={
      Object {
        "borderBottomRightRadius": 3,
        "borderLeftColor": "red",
        "borderLeftWidth": 300,
        "borderTopRightRadius": 3,
        "flex": 1,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      disabled={false}
      onPress={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
      underlayColor="#2089dc"
    >
      <Component
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View />
      </Component>
    </TouchableOpacity>
  </Component>
</Component>
`;

exports[`ButtonGroup Component should render without inner borders 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#fff",
      "borderColor": "#e3e3e3",
      "borderRadius": 3,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 40,
      "marginBottom": 5,
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 5,
      "overflow": "hidden",
    }
  }
>
  <Component
    key="0"
    style={
      Object {
        "borderBottomLeftRadius": 3,
        "borderRightColor": "#bdc6cf",
        "borderRightWidth": 0,
        "borderTopLeftRadius": 3,
        "flex": 1,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      disabled={false}
      onPress={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
      underlayColor="#2089dc"
    >
      <Component
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Themed.Text
          style={
            Object {
              "color": "#5e6977",
              "fontSize": 16.25,
              "fontWeight": "500",
            }
          }
          testID="buttonGroupItemText"
        >
          Button 1
        </Themed.Text>
      </Component>
    </TouchableOpacity>
  </Component>
  <Component
    key="1"
    style={
      Object {
        "borderLeftColor": "#bdc6cf",
        "borderLeftWidth": 0,
        "borderRightColor": "#bdc6cf",
        "borderRightWidth": 0,
        "flex": 1,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      disabled={false}
      onPress={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
      underlayColor="#2089dc"
    >
      <Component
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Themed.Text
          style={
            Object {
              "color": "#5e6977",
              "fontSize": 16.25,
              "fontWeight": "500",
            }
          }
          testID="buttonGroupItemText"
        >
          Button 2
        </Themed.Text>
      </Component>
    </TouchableOpacity>
  </Component>
  <Component
    key="2"
    style={
      Object {
        "borderBottomRightRadius": 3,
        "borderTopRightRadius": 3,
        "flex": 1,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      disabled={false}
      onPress={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
      underlayColor="#2089dc"
    >
      <Component
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Themed.Text
          style={
            Object {
              "color": "#5e6977",
              "fontSize": 16.25,
              "fontWeight": "500",
            }
          }
          testID="buttonGroupItemText"
        >
          Button 3
        </Themed.Text>
      </Component>
    </TouchableOpacity>
  </Component>
</Component>
`;

exports[`ButtonGroup Component should render without issues 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#fff",
      "borderColor": "#e3e3e3",
      "borderRadius": 3,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 40,
      "marginBottom": 5,
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 5,
      "overflow": "hidden",
    }
  }
>
  <Component
    key="0"
    style={
      Object {
        "borderBottomLeftRadius": 3,
        "borderRightColor": "#bdc6cf",
        "borderRightWidth": 0,
        "borderTopLeftRadius": 3,
        "flex": 1,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      disabled={false}
      onPress={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
      underlayColor="#2089dc"
    >
      <Component
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Themed.Text
          style={
            Object {
              "color": "#5e6977",
              "fontSize": 16.25,
              "fontWeight": "500",
            }
          }
          testID="buttonGroupItemText"
        >
          Button 1
        </Themed.Text>
      </Component>
    </TouchableOpacity>
  </Component>
  <Component
    key="1"
    style={
      Object {
        "borderLeftColor": "#bdc6cf",
        "borderLeftWidth": 1,
        "borderRightColor": "#bdc6cf",
        "borderRightWidth": 1,
        "flex": 1,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      disabled={false}
      onPress={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
      underlayColor="#2089dc"
    >
      <Component
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Themed.Text
          style={
            Object {
              "color": "#5e6977",
              "fontSize": 16.25,
              "fontWeight": "500",
            }
          }
          testID="buttonGroupItemText"
        >
          Button 2
        </Themed.Text>
      </Component>
    </TouchableOpacity>
  </Component>
  <Component
    key="2"
    style={
      Object {
        "borderBottomRightRadius": 3,
        "borderTopRightRadius": 3,
        "flex": 1,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      disabled={false}
      onPress={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
      underlayColor="#2089dc"
    >
      <Component
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Themed.Text
          style={
            Object {
              "color": "#5e6977",
              "fontSize": 16.25,
              "fontWeight": "500",
            }
          }
          testID="buttonGroupItemText"
        >
          Button 3
        </Themed.Text>
      </Component>
    </TouchableOpacity>
  </Component>
</Component>
`;
